test/e2e-olm/utils.go

```diff
@@ -1,232 +1,301 @@
+		Spec: loggingv1.ElasticsearchSpec{
+			Spec: loggingv1.ElasticsearchNodeSpec{
 				Image: "",
-				Resources: v1.ResourceRequirements{
-					Limits: v1.ResourceList{
-						v1.ResourceMemory: memValue,
+				Resources: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceMemory: memValue,
 					},
-					Requests: v1.ResourceList{
-						v1.ResourceCPU:    cpuValue,
-						v1.ResourceMemory: memValue,
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    cpuValue,
+						corev1.ResourceMemory: memValue,
 					},
 				},
 			},
-			Nodes: []elasticsearch.ElasticsearchNode{
+			Nodes: []loggingv1.ElasticsearchNode{
 				esDataNode,
 			},
-			ManagementState:  elasticsearch.ManagementStateManaged,
-			RedundancyPolicy: elasticsearch.ZeroRedundancy,
+			ManagementState:  loggingv1.ManagementStateManaged,
+			RedundancyPolicy: loggingv1.ZeroRedundancy,
 		},
 	}
 
 	t.Logf("Creating Elasticsearch CR: %v", cr)
-	err = f.Client.Create(goctx.TODO(), cr, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+
+	cleanupOpts := &test.CleanupOptions{
+		TestContext:   ctx,
+		Timeout:       cleanupTimeout,
+		RetryInterval: cleanupRetryInterval,
+	}
+
+	err = f.Client.Create(context.TODO(), cr, cleanupOpts)
 	if err != nil {
 		return nil, fmt.Errorf("could not create exampleElasticsearch: %v", err)
 	}
 
 	return cr, nil
 }
 
+func updateElasticsearchSpec(t *testing.T, f *test.Framework, desired *loggingv1.Elasticsearch) error {
+	return wait.Poll(retryInterval, timeout, func() (bool, error) {
+		current := &loggingv1.Elasticsearch{}
+		key := client.ObjectKey{Name: desired.GetName(), Namespace: desired.GetNamespace()}
+
+		if err := f.Client.Get(context.TODO(), key, current); err != nil {
+			if errors.IsNotFound(err) {
+				// Stop retry because CR not found
+				return false, err
+			}
+
+			// Transient error retry
+			return false, nil
+		}
+
+		current.Spec = desired.Spec
+
+		t.Logf("Update Spec: %#v", current.Spec)
+
+		if err := f.Client.Update(context.TODO(), current); err != nil {
+			if errors.IsConflict(err) {
+				// Retry update because resource needs to get updated
+				return false, nil
+			}
+
+			// Stop retry not recoverable error
+			return false, err
+		}
+
+		return true, nil
+	})
+}
+
 // Create the secret that would be generated by CLO normally
-func createElasticsearchSecret(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+func createElasticsearchSecret(t *testing.T, f *test.Framework, ctx *test.Context, uuid string) error {
 	t.Log("Creating required secret")
-	namespace, err := ctx.GetNamespace()
+	namespace, err := ctx.GetWatchNamespace()
 	if err != nil {
 		return fmt.Errorf("Could not get namespace: %v", err)
 
 	}
 
+	if err := generateCertificates(t, namespace, uuid); err != nil {
+		return err
+	}
+
 	elasticsearchSecret := utils.Secret(
-		elasticsearchCRName,
+		elasticsearchNameFor(uuid),
```

Comment:
the secret name needs to match the CR name -- it has nothing to do with the UUID